app/models/game.rb

```diff
@@ -13,6 +13,25 @@ class Game < ActiveRecord::Base
 
   scope :available, -> { where('black_player_id IS NULL OR white_player_id IS NULL') }
 
+  def check?
+    #game is in check if either black or white king is in check
+    king_is_in_check?('black') || king_is_in_check('white')
+  end
+
+  def king_is_in_check?(color)
+    #finds king by color
+    king = pieces.find_by(type: "King", color: color)
+    #returns array of all opposing pieces
+    opponents = pieces.includes(:game).where(color: !king.color).to_a
```

Comment:
I think what you want here is

`pieces.where.not(color: color)`

The `to_a` shouldn't be needed because ActiveRecord will already return an `enumerable`.

Also `includes(:game)` shouldn't be needed because we've already loaded the current game.